src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureKeyVaultReference/AzureKeyVaultSecretProvider.cs

```diff
@@ -63,6 +82,38 @@ public async Task<string> GetSecretValue(Uri secretUri, CancellationToken cancel
             return secretValue;
         }
 
+        internal bool AnyExpiredSecrets()
+        {
+            bool shouldRefreshKeyVaultSecrets = false;
+            List<CachedKeyVaultSecret> secretsToBeRemovedFromCache = new List<CachedKeyVaultSecret>();
+
+            foreach (var cachedSecret in _cachedKeyVaultSecrets)
+            {
+                // Skip the refresh for this key vault secret if it has no expiration time or if it hasn't expired yet
+                if (cachedSecret.ExpiresOn == null || DateTimeOffset.UtcNow < cachedSecret.ExpiresOn)
+                {
+                    continue;
+                }
+
+                // Remove the cached Key Vault secret for this key
+                secretsToBeRemovedFromCache.Add(new CachedKeyVaultSecret(cachedSecret.Key));
+                shouldRefreshKeyVaultSecrets = true;
+            }
+
+            secretsToBeRemovedFromCache.ForEach(secret => RemoveExpiredSecretFromCache(secret.Key));
+            return shouldRefreshKeyVaultSecrets;
+        }
+
+        internal void RemoveAllSecretsFromCache()
+        {
+            _cachedKeyVaultSecrets.Clear();
+        }
+
+        internal void RemoveExpiredSecretFromCache(string key)
```

Comment:
>RemoveExpiredSecretFromCache [](start = 22, length = 28)

That is a bit specific of a name. Although we are using it for expired secrets it's really just removing something from the cache. Can be `SetDirty(string key)`

Reply:
+1, maybe `RemoveSecretFromCache`.

Reply:
Updated to `RemoveSecretFromCache`